src/main/java/uk/gov/hmcts/reform/adoption/service/CoreCaseDataService.java

```diff
@@ -0,0 +1,54 @@
+package uk.gov.hmcts.reform.adoption.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.adoption.request.RequestData;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.ccd.client.CoreCaseDataApi;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDataContent;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.ccd.client.model.Event;
+import uk.gov.hmcts.reform.ccd.client.model.StartEventResponse;
+import uk.gov.hmcts.reform.idam.client.IdamClient;
+import uk.gov.hmcts.reform.idam.client.models.UserDetails;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class CoreCaseDataService {
+    private final CoreCaseDataApi coreCaseDataApi;
+    private final AuthTokenGenerator authTokenGenerator;
+    private final IdamClient idamClient;
+    private final RequestData requestData;
+
+    static final String JURISDICTION = "ADOPTION";
+    static final String CASE_TYPE = "ADOPTION";
+    static final String START_CASE_EVENT = "openCase";
+
+    public CaseDetails startCase() {
+        String authorisation = requestData.authorisation();
+        UserDetails userDetails = idamClient.getUserDetails(authorisation);
```

Comment:
do we need this call, it is used to get userId, is it not same as in requestData.userId()

Reply:
yes, if we always send auth and userId back in response from front end this is fine

Reply:
have removed the call to idam and all imports for it